src/components/RestaurantList.tsx

```diff
@@ -1,65 +1,53 @@
 import { Category, Restaurant, SortingType } from '../types/restaurant';
 
-import React, { ChangeEvent } from 'react';
-import { RestaurantItem } from './';
+import React, { MouseEvent, ChangeEvent, useState } from 'react';
+import { Select, RestaurantItem } from './';
 import styled from 'styled-components';
 
+import filterRestaurants from '../domain/filterRestaurants';
 import { CATEGORIES, SORTING_TYPES } from '../constants';
 
 interface Props {
   restaurants: Restaurant[];
   openModal: (id: Restaurant['id']) => void;
-  setCategory: (category: Category) => void;
-  setSortingType: (sortingType: SortingType) => void;
 }
 
-class RestaurantList extends React.Component<Props> {
-  openDetailModal = (e: any) => {
-    const { id } = e.target.closest('li');
-    this.props.openModal(id);
-  };
+const RestaurantList = ({ restaurants, openModal }: Props) => {
+  const [category, setCategory] = useState<Category | '전체'>('전체');
+  const [sortingType, setSortingType] = useState<SortingType>('이름순');
+  const filteredRestaurants = filterRestaurants(restaurants, category, sortingType);
 
-  onChangeCategory = (e: ChangeEvent) => {
-    if (!(e.target instanceof HTMLSelectElement)) return;
+  const openDetailModal = (e: MouseEvent) => {
+    if (!(e.target instanceof HTMLElement)) return;
 
-    const { value: category } = e.target;
-    this.props.setCategory(category as Category);
+    const $li = e.target.closest('li');
+    if ($li) openModal($li.id as Restaurant['id']);
   };
 
-  onChangeSortingType = (e: ChangeEvent) => {
+  const onChangeSelect = (e: ChangeEvent) => {
     if (!(e.target instanceof HTMLSelectElement)) return;
-
-    const { value: sortingType } = e.target;
-    this.props.setSortingType(sortingType as SortingType);
+    const { name, value } = e.target;
```

Comment:
![image](https://user-images.githubusercontent.com/112997662/232265473-7cd5a6eb-9142-444a-ac74-9e668f2e49ec.png)

openDetailModal처럼 해당 코드 윗줄에 개행을 주면 좀 더 속이 시원할 것 같아요

```suggestion
    if (!(e.target instanceof HTMLSelectElement)) return;
//////////////////////// 개행
    const { name, value } = e.target;
```
![image](https://user-images.githubusercontent.com/112997662/232265457-b0408629-2d1b-4bec-867b-2b627d73ea4e.png)